roomserver/internal/input/input_events.go

```diff
@@ -113,11 +172,37 @@ func (r *Inputer) processRoomEvent(
 		// current room state.
 		softfail, err = helpers.CheckForSoftFail(ctx, r.DB, headered, input.StateEventIDs)
 		if err != nil {
-			logrus.WithFields(logrus.Fields{
-				"event_id": event.EventID(),
-				"type":     event.Type(),
-				"room":     event.RoomID(),
-			}).WithError(err).Info("Error authing soft-failed event")
+			logger.WithError(err).Info("Error authing soft-failed event")
+		}
+	}
+
+	missingPrev := !input.HasState && len(missingRes.MissingPrevEventIDs) > 0
```

Comment:
Needs comments. Specifically, if we _are_ missing prev events so `len(missingRes.MissingPrevEventIDs) > 0` but we `HasState`, why does this get out of the following code? It's not clear why this makes sense.